src/GslibIO.jl

```diff
@@ -44,6 +52,25 @@ function load(file::File{format"GSLIB"})
   end
 end
 
+# helper function to parse a legacy GSLIB file
+function parse_legacy(filename::AbstractString)
+  open(filename) do fs
+    header = readline(fs)
+    # the second line contains the number of variables (the first integer)
+    # and it may contain extra info (comments), which will be ignored
+    linesplit = split(readline(fs))
+    nvars = parse(Int, linesplit[1])
+    varnames = [Symbol(strip(readline(fs))) for i in 1:nvars]
+
+    metadata = LegacySpec(header, nvars, varnames)
```

Comment:
let's rename this to `meta` so that it is the same length of the other variable in the returned pair `data`.

Reply:
also it feels a bit strange to return `data` separately instead of making it a fourth field of the LegacySpec struct. can you make the `data` table a fourth field of the spec? So this parser would return a single object with the full specification.

Reply:
> let's rename this to `meta` so that it is the same length of the other variable in the returned pair `data`.

Meta does not mean anything. Meta-what. I suggest to keep it as metadata.

Reply:
it is ok to leave it as metadata, I just feel that these are extra characters to type, and we know what we mean when we type the shorter version in this context. in any case, I think we need to put the data inside the struct to return a single complete specification.

Reply:
> also it feels a bit strange to return `data` separately instead of making it a fourth field of the LegacySpec struct. can you make the `data` table a fourth field of the spec? So this parser would return a single object with the full specification.

In my opinion it is not big deal to have an structure mixing the metadata and data, but more confusing. However, I can put everything together for simplicity, but I think it is not good idea mixing both concepts, especially if they are used for an internal implementation. We are not returning the metadata to the caller.

Reply:
We are returning the full spec which is clearer in my opinion. The caller can then read the spec struct to understand what it contains, and then access `spec.data` if needed.

Reply:
> it is ok to leave it as metadata, I just feel that these are extra characters to type, and we know what we mean when we type the shorter version in this context. in any case, I think we need to put the data inside the struct to return a single complete specification.

Explicit is better than implicit :-)

Reply:
> I think we need to put the data inside the struct to return a single complete specification

OK, but we would need to change the semantic of LegacySpec, because it will not be anymore just the specification, but the specification and the content (data). Any suggestion for a new name for it?

Reply:
The data **is** part of the specification, I don't understand what you are saying.

Reply:
The GSLIB spec is a spec for the entire file, which contains a header with a specific format plus a data table. Just add the field to the struct and we can avoid returning two arguments when we could return just one. This also helps with piping commands in general. Returning more than one thing leads to complications downstream if you want to `a |> foo |> bar |> ...`

Reply:
It is just a different point of view, which is not very relevant. In summary, let's add the data to the structure as well.